app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java

```diff
@@ -172,40 +199,88 @@ private void setInfo(final StreamInfo info) {
     }
 
     public void setAudioStreams(final List<AudioStream> audioStreams) {
+        removeDeliveryDistinctStreams(audioStreams, DeliveryMethod.PROGRESSIVE_HTTP);
         setAudioStreams(new StreamSizeWrapper<>(audioStreams, getContext()));
     }
 
     public void setAudioStreams(final StreamSizeWrapper<AudioStream> was) {
         this.wrappedAudioStreams = was;
```

Comment:
Actually, this method returns only streams of the delivery method passed as the second parameter.

Here this is intended, as the downloader only supports progressive streams.

Reply:
Yeah, sorry, I meant:

Why in this case the **non-**`PROGRESSIVE_HTTP` streams are not removed?

Reply:
Because this method is only called by `setAudioStreams(List<AudioStream> audioStreams)`. That's the same behavior with the corresponding video and subtitle methods.

By finding this, I also found that the `removeDeliveryDistinctStreams` method is called twice, for audio, video and subtitle streams. I will change this.

Reply:
Done